packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -417,26 +462,45 @@ public void onPathReady(String path) {
     finishWithSuccess(null);
   }
 
-  private void handleImageResult(String path, boolean shouldDeleteOriginalIfScaled) {
+  private void handleMultiImageResult(
+      ArrayList<String> paths, boolean shouldDeleteOriginalIfScaled) {
     if (methodCall != null) {
-      Double maxWidth = methodCall.argument("maxWidth");
-      Double maxHeight = methodCall.argument("maxHeight");
-      Integer imageQuality = methodCall.argument("imageQuality");
-
-      String finalImagePath =
-          imageResizer.resizeImageIfNeeded(path, maxWidth, maxHeight, imageQuality);
-
-      finishWithSuccess(finalImagePath);
+      for (int i = 0; i < paths.size(); i++) {
+        String finalImagePath = resizeImage(paths.get(i));
+
+        //delete original file if scaled
+        if (finalImagePath != null
+            && !finalImagePath.equals(paths.get(i))
+            && shouldDeleteOriginalIfScaled) {
+          new File(paths.get(i)).delete();
+        }
+        paths.set(i, finalImagePath);
+      }
+      finishWithListSuccess(paths);
+    }
+  }
 
+  private void handleImageResult(String path, boolean shouldDeleteOriginalIfScaled) {
+    if (methodCall != null) {
+      String finalImagePath = resizeImage(path);
       //delete original file if scaled
       if (finalImagePath != null && !finalImagePath.equals(path) && shouldDeleteOriginalIfScaled) {
         new File(path).delete();
       }
+      finishWithSuccess(finalImagePath);
     } else {
       finishWithSuccess(path);
     }
   }
 
+  private String resizeImage(String path) {
```

Comment:
Since the param and return types are both string, the name "resizeImage" is a little confusing. 
Maybe 
```java
private String getResizedImagePath(String originalImagePath, Double maxWidth, Double maxHeight, Integer imageQuality)
```

Reply:
Done.